### PR TITLE
added noa system prompt 

### PR DESCRIPTION
**Pull request by uma-shankar-TE**
This will be used as additional instructions set by user from Noa app on top of system prompt

### CODE REVIEW

File: assistant/gpt_assistant.py [hunk 1]
```diff
@@ -416,7 +417,10 @@ async def send_to_assistant(
         # Inject context into our copy by appending it to system message. Unclear whether multiple
         # system messages are confusing to the assistant or not but cursory testing shows this
         # seems to work.
-        extra_context_message = Message(role=Role.SYSTEM, content=self._create_context_system_message(local_time=local_time, location=location_address, learned_context=learned_context))
+        additional_context = self._create_context_system_message(local_time=local_time, location=location_address, learned_context=learned_context)
+        if noa_system_prompt is not None:
+            additional_context = f"{noa_system_prompt}\n{additional_context}"
+        extra_context_message = Message(role=Role.SYSTEM, content=additional_context)
```
Comment (trzy) [inline assistant/gpt_assistant.py]: How long are these prompts expected to be? Should they be appended as an extra system message or in the primary system message? Note that Claude does not allow multiple system messages.

File: assistant/gpt_assistant.py [hunk 2]
```diff
@@ -373,6 +373,7 @@ def __init__(self, client: openai.AsyncOpenAI):
     async def send_to_assistant(
         self,
         prompt: str,
+        noa_system_prompt: str|None,
```
Comment (trzy) [inline assistant/gpt_assistant.py]: Please add whitespace to match the formatting of the other params.

File: app.py [hunk 3]
```diff
@@ -56,15 +56,15 @@ def __call__(self, data: str = Form(...)):
                 status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
             )
 
-def transcribe(client: openai.OpenAI, audio_bytes: bytes) -> str:
+async def transcribe(client: openai.OpenAI, audio_bytes: bytes) -> str:
```
Comment (trzy) [inline app.py]: openai.OpenAI -> openai.AsyncOpenAI

**Review by trzy — CHANGES_REQUESTED** [review]

File: assistant/gpt_assistant.py (hunk 1 above)
Comment (uma-shankar-TE) [reply assistant/gpt_assistant.py]: Its for Noa app specific like taking a subtle character.
for claude we can append in system prompt

File: app.py (hunk 3 above)
Comment (uma-shankar-TE) [reply app.py]: done https://github.com/brilliantlabsAR/noa-assistant/pull/9/commits/86b7b9bdb3f875301f2412a1b52be7446164c1a4

File: assistant/gpt_assistant.py (hunk 2 above)
Comment (uma-shankar-TE) [reply assistant/gpt_assistant.py]: done https://github.com/brilliantlabsAR/noa-assistant/pull/9/commits/86b7b9bdb3f875301f2412a1b52be7446164c1a4

**Pull request merged**